rtcrtpreceiver.go

```diff
@@ -1,14 +1,117 @@
 package webrtc
 
+import (
+	"github.com/pions/webrtc/pkg/rtcp"
+	"github.com/pions/webrtc/pkg/rtp"
+)
+
 // RTCRtpReceiver allows an application to inspect the receipt of a RTCTrack
 type RTCRtpReceiver struct {
+	kind      RTCRtpCodecType
+	transport *RTCDtlsTransport
+
+	hasRecv chan bool
+
 	Track *RTCTrack
-	// receiverTrack *RTCTrack
-	// receiverTransport
-	// receiverRtcpTransport
+
+	rtpOut  chan *rtp.Packet
+	rtcpOut chan rtcp.Packet
+}
+
+// NewRTCRtpReceiver constructs a new RTCRtpReceiver
+func NewRTCRtpReceiver(kind RTCRtpCodecType, transport *RTCDtlsTransport) *RTCRtpReceiver {
+	return &RTCRtpReceiver{
+		kind:      kind,
+		transport: transport,
+		hasRecv:   make(chan bool),
+		rtpOut:    make(chan *rtp.Packet, 15),
+		rtcpOut:   make(chan rtcp.Packet, 15),
+	}
 }
 
-// TODO: receiving side
-// func newRTCRtpReceiver(kind, id string) {
-//
-// }
+// Receive blocks until the RTCTrack is available
+func (r *RTCRtpReceiver) Receive(parameters RTCRtpReceiveParameters) chan bool {
+	// TODO atomic only allow this to fire once
```

Comment:
Will grab this with https://github.com/pions/webrtc/issues/106

We have added a lot of TODOs, I am just going to make a monolithic ticket.